odbc/src/api/data.rs

```diff
@@ -1725,4 +1859,341 @@ mod unit {
             );
         }
     }
+
+    mod convert_to_datetime {
```

Comment:
I wonder if you could simplify the way you process all these tests, similar to how I did in #36 and how I'm doing in #37 . You may be able to use the same macros.

Reply:
yeah, lookin closely (especially at the `test_conversion_ok` and `test_conversion_err` macros) I think those do exactly what the functions I have written do! So should generalize well. Just to make sure I'm on the same page, are you open to expanding the `conversion` mod to include dates/times?

Reply:
Absolutely.